### DISCUSSION TITLE
Mol File renders image as Unexpected.

### DISCUSSION THREAD

**Discussion by Suganesh-repo** [rdkit/rdkit#7317] [Q&A]
for this Mol3000 file, 

2244
     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 13 13 0 0 0
M  V30 BEGIN ATOM
M  V30 1 O 3.000000 0.000000 0.000000 0
M  V30 2 O 0.750000 -3.897100 0.000000 0
M  V30 3 O 13.521220 -4.404864 0.000000 0
M  V30 4 O 3.000000 2.598100 0.000000 0
M  V30 5 C 1.500000 0.000000 0.000000 0
M  V30 6 C 0.750000 -1.299000 0.000000 0
M  V30 7 C 0.750000 1.299000 0.000000 0
M  V30 8 C -0.750000 -1.299000 0.000000 0
M  V30 9 C -0.750000 1.299000 0.000000 0
M  V30 10 C -1.500000 0.000000 0.000000 0
M  V30 11 C 1.500000 -2.598100 0.000000 0
M  V30 12 C 3.750000 1.299000 0.000000 0
M  V30 13 C 5.250000 1.299000 0.000000 0
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 1 5
M  V30 2 1 1 12
M  V30 3 1 2 11
M  V30 4 2 3 11
M  V30 5 2 4 12
M  V30 6 2 5 6
M  V30 7 1 5 7
M  V30 8 1 6 8
M  V30 9 1 6 11
M  V30 10 2 7 9
M  V30 11 2 8 10
M  V30 12 1 9 10
M  V30 13 1 12 13
M  V30 END BOND
M  V30 END CTAB
M  END

<img width="674" alt="Screenshot 2024-03-31 at 11 22 48 AM" src="https://github.com/rdkit/rdkit/assets/99866614/07b93d7c-f86f-4d75-b767-ddc7faefd4f4">

aspirin is like this, the bond length should be lengthy. 

the below image is from ketcher, it renders the image as expected..
<img width="434" alt="Screenshot 2024-03-31 at 11 25 42 AM" src="https://github.com/rdkit/rdkit/assets/99866614/828237a6-e403-44d9-b04e-3f6a5e87edf0">

**Comment by DavidACosgrove** [comment]
Hi @Suganesh-repo,
Can you please repost this with the mol block between backticks so that the formatting of the file is preserved.  The file isn't readable as it is.
Dave

**Comment by Suganesh-repo** [comment]
`2244
     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 13 13 0 0 0
M  V30 BEGIN ATOM
M  V30 1 O 3.000000 0.000000 0.000000 0
M  V30 2 O 0.750000 -3.897100 0.000000 0
M  V30 3 O 8.510836 -3.131853 0.000000 0
M  V30 4 O 3.000000 2.598100 0.000000 0
M  V30 5 C 1.500000 0.000000 0.000000 0
M  V30 6 C 0.750000 -1.299000 0.000000 0
M  V30 7 C 0.750000 1.299000 0.000000 0
M  V30 8 C -0.750000 -1.299000 0.000000 0
M  V30 9 C -0.750000 1.299000 0.000000 0
M  V30 10 C -1.500000 0.000000 0.000000 0
M  V30 11 C 1.500000 -2.598100 0.000000 0
M  V30 12 C 3.750000 1.299000 0.000000 0
M  V30 13 C 5.250000 1.299000 0.000000 0
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 1 5
M  V30 2 1 1 12
M  V30 3 1 2 11
M  V30 4 2 3 11
M  V30 5 2 4 12
M  V30 6 2 5 6
M  V30 7 1 5 7
M  V30 8 1 6 8
M  V30 9 1 6 11
M  V30 10 2 7 9
M  V30 11 2 8 10
M  V30 12 1 9 10
M  V30 13 1 12 13
M  V30 END BOND
M  V30 END CTAB
M  END
`

**Reply by DavidACosgrove** [reply]
Sorry, @Suganesh-repo, it needs to be between triple backticks, as if it was a bit of code, as described at https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.  You can use the preview tab in GitHub to check whether what will be displayed is the same as it appears in the file.  Molblocks have a fixed format and without the spaces and characters in exactly the right place the file won't be read correctly.  It would also be useful if you could post a small piece of code that generates the image, so we can see what drawing options you used.

**Reply by Suganesh-repo** [reply]
```
2244
     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 13 13 0 0 0
M  V30 BEGIN ATOM
M  V30 1 O 2.598085 -1.499986 0.000000 0
M  V30 2 O -1.299010 -3.749995 0.000000 0
M  V30 3 O 9.923686 -6.081460 0.000000 0
M  V30 4 O 3.897122 0.750042 0.000000 0
M  V30 5 C 1.299042 -0.749993 0.000000 0
M  V30 6 C 0.000027 -1.499967 0.000000 0
M  V30 7 C 1.299015 0.749974 0.000000 0
M  V30 8 C -1.299015 -0.749974 0.000000 0
M  V30 9 C -0.000027 1.499967 0.000000 0
M  V30 10 C -1.299042 0.749993 0.000000 0
M  V30 11 C 0.000005 -3.000021 0.000000 0
M  V30 12 C 3.897099 -0.750011 0.000000 0
M  V30 13 C 5.196142 -1.500004 0.000000 0
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 1 5
M  V30 2 1 1 12
M  V30 3 1 2 11
M  V30 4 2 3 11
M  V30 5 2 4 12
M  V30 6 2 5 6
M  V30 7 1 5 7
M  V30 8 1 6 8
M  V30 9 1 6 11
M  V30 10 2 7 9
M  V30 11 2 8 10
M  V30 12 1 9 10
M  V30 13 1 12 13
M  V30 END BOND
M  V30 END CTAB
M  END

```

JS CODE
```
 var params = JSON.stringify({'sanitize':false, 'kekulize':false});
 var mol = RDKitModule.get_mol(molblock, params);
 var dopt = {"useMolBlockWedging":true,"addStereoAnnotation":true,"setBackgroundColour":"(0,.9,.9,.3)","useComplexQueryAtomSymbols":true,"includeRadicals":true,"addAtomIndices":true,"explicitMethyl":true,"fixedBondLength":30,"noAtomLabels":false,'width':500,'height':500,'rotate':0}; 
 var svg = mol.get_svg_with_highlights(JSON.stringify(dopt));
```

and this svg is like the above, disconnected bond.
 
 
 Pls do check now @DavidACosgrove . Thanks in advance

**Comment by DavidACosgrove** [comment]
@Suganesh-repo Great thanks.  I've just made a PR that fixes this.
@greglandrum This was a discussion, not a bug report.  Will that make a difference anywhere in the GitHub process?

**Reply by Suganesh-repo** [reply]
Thanks @DavidACosgrove . May i know when will it get merge into master branch?

**Reply by greglandrum** [reply]
it will be merged after I've had a chance to review it.

**Reply by Suganesh-repo** [reply]
@greglandrum  May I know if this fix merged into the master branch?
